tensorflow/python/client/session_clusterspec_prop_test.py

```diff
@@ -323,8 +325,12 @@ def testClusterSpecPropagationThreeServers(self):
     job2.tasks[0] = server2.target[len('grpc://'):]
     job2.tasks[1] = server3.target[len('grpc://'):]
 
-    config1 = config_pb2.ConfigProto(cluster_def=cluster_def1)
-    config2 = config_pb2.ConfigProto(cluster_def=cluster_def2)
+    config1 = config_pb2.ConfigProto(
+        cluster_def=cluster_def1,
+        isolate_session_state=True)
+    config2 = config_pb2.ConfigProto(
+        cluster_def=cluster_def2,
+        isolate_session_state=True)
```

Comment:
If I'm reading this test correctly, it looks like the default will change from previously defaulting to `isolate_session_state=True` to defaulting to `isolate_session_state=False`. Is this true? If so, can we think of a way to avoid this surprising change in behavior?

Reply:
@saeta You are right. This incompatibility has embarrassed me a lot :(

AFAIK, in ProtoBuf 3, there is no way to test whether a field is explicitly set or net. Thus, if we set the default value of `isolate_session_state` to be `false`, it will bring another incompatibility.

How about adding a new filed `share_session_state` defaulting to `false` in `tf.ConfigProto` and marking `isolate_session_state` to be deprecated? Then:
- If `share_session_state` is not set, no change happens.
- If `share_session_state` is turned on, a GRPC server will offer renamed devices to its created worker sessions.

Looking forward to your suggestions.

Reply:
Done with a new [commit](2bf038739467f5131dd98b91d66747aae6c89972) which adds a new field `share_session_state` in `tf.ConfigProto`.

@saeta @mrry Is this change enough for keeping compatibility? Thanks a lot.